temaRefRadoiOlimpiu/nrPrime.py

```diff
@@ -0,0 +1,30 @@
+#Print all prime numbers in the user-specified range.
+#A number is prime if it is divisible only by itself and by one.
+#For instance, 3 is a prime number, but 4 is not. 
+
+
+arr = [0] * 100
+n = int(input())
+for i in range (0,n):
+    arr[i] = int(input())
+
+
+for i in range (0,n):
+    prim = True
+    d = 2
+    while d<=int(arr[i]/2) and prim:

```

Comment:
impartirea se face in afara conditiei whileului pentru a nu o calcula la fiecare iteratie. avand acelasi rezultat nu are rost sa imparti arr[i] la 2 ca sa verifici dacai mai ai d-uri